### PR TITLE
update periodic_test.go & run unit tests with -race param

### PR DESCRIPTION
**Pull request by tremes**
Updates `periodic_test.go` to be more robust and readable and also adds `-race` param to unit tests make target.

## Categories

- [ ] Bugfix
- [ ] Data Enhancement
- [ ] Feature
- [ ] Backporting
- [X] Others (CI, Infrastructure, Documentation)

## Sample Archive

- no new data

## Unit Tests

- `pkg/controller/periodic/periodic_test.go`

## Privacy

Yes. There are no sensitive data in the newly collected information.

## Breaking Changes

Yes/No

## References

https://issues.redhat.com/browse/???
https://access.redhat.com/solutions/???

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -58,7 +58,7 @@ test: unit ## Run all the tests
 # Run the unit tests
 .PHONY: unit
 unit: ## Run the unit tests
-	go test $(GO_TEST_FLAGS) -coverprofile $(COVER_PROFILE) ./...
+	go test -race $(GO_TEST_FLAGS) -coverprofile $(COVER_PROFILE) ./...
```
Comment (tremes) [inline Makefile:61]: I am still wondering if we should also add e.g `-count 20`, which should help detect possible concurrency problems faster (if I am not mistaken), but the tests will also take longer to run.

Comment (tremes) [inline Makefile:61]: I am still wondering if we should also add e.g `-count 20`, which should help detect possible concurrency problems faster (if I am not mistaken), but the tests will also take longer to run.

Comment (ncaak) [reply Makefile:61]: Did you do a timed test on that to see how long could it take?

Comment (tremes) [reply Makefile:61]: I guess 20x longer.....

**Pull request merged**